### ISSUE TITLE
Calva shows two hover definitions for user-defined vars when a repl is connected

### ISSUE THREAD

**Issue by bpringe** [BetterThanTomorrow/calva#2091]
![image](https://user-images.githubusercontent.com/12722744/221080897-d1a37a50-9b1b-4fcc-88d7-a3325d223c98.png)

It doesn't seem like this happens for core functions.

I haven't yet tried previous Calva versions.

**Comment by julienvincent** [comment]
I know this happens when there are two LSP clients running that are both governing the same file. Can you confirm if this is the case for your project, or are you sure this is a result of the REPL connection?

**Comment by PEZ** [comment]
We have some checks for if the REPL has an answer there, iirc. So it could be (my guess it is) to do with several lsp servers serving the same file, _and_ be triggered when there is a REPL connection.

**Comment by bpringe** [comment]
I've verified that there's only one clojure-lsp process running when I see the double hover entries. Also, clojure-lsp hover entries include the file path, and the repl ones don't. There's one of each, as you can see in the image.

**Comment by bpringe** [comment]
Another thing to note is the difference in doc strings in the original image. I just did another test to verify that one entry is from the repl.

![image](https://user-images.githubusercontent.com/12722744/221377578-a2817f99-c954-4207-b0e2-28eda00e6bc8.png)

I added content to the doc string, evaluated the function, then removed the content without evaluating the function again. The repl entry still shows the evaluated version of the doc string (bottom), but clojure-lsp shows the current version in the file (top), as expected.
